cocos/renderer/backend/ProgramCache.cpp

```diff
@@ -303,6 +305,29 @@ void ProgramCache::removeAllPrograms()
         program.second->release();
     }
     _cachedPrograms.clear();
+    
+    for (auto& program : _cachedCustomPrograms)
+    {
+        program.second->release();
+    }
+    _cachedCustomPrograms.clear();
+}
+
+void ProgramCache::addCustomProgram(const std::string &key, backend::Program *program)
+{
+    _cachedCustomPrograms.emplace(key, program);
```

Comment:
need release old program if key is already exist in `_cachedCustomPrograms`.

Reply:
I have fixed this in https://github.com/ironhidegames/cocos2d-x/commit/69f393333a72d1956f589199209069637a157067
Should I make a new PR?

Reply:
Yes, please.

Reply:
Yes, It would be great.

Reply:
I have made a new PR: https://github.com/cocos2d/cocos2d-x/pull/20550